### PR TITLE
2816: Fix Linux CI Launch Test Failure

### PR DESCRIPTION
**Pull request by krzywon**
## Description

This fixes the issue that prevented the sasview app from launching during the Linux CI test steps.

IMPORTANT NOTES:
 - This does not solve the underlying issue where the CI was labelled successful despite the failure to launch. This refers to #2816, but the issue should remain open for now until a solution can be found to capture the failure..
 - This branch has the work from #2862. Without these changes, the linux sasview app cannot launch. The changes to `.github/workflows/ci.yml` are the only unique changes made to this branch.

## How Has This Been Tested?

CI - Success!

## Review Checklist:

**Documentation** (check at least one)
- [x] There is **nothing** that needs documenting
- [ ] Documentation changes are **in this PR**
- [ ] There is an **issue** open for the documentation (link?)

**Installers**
- [x] There is a chance this will affect the **installers**, if so
  - [ ] **Ubuntu** installer (GH artifact) has been tested (installed and worked) 

**Licencing** (untick if necessary)
- [x] The introduced changes comply with SasView license (BSD 3-Clause)

### CODE REVIEW

File: src/sas/qtgui/Utilities/DocViewWidget.py [hunk 1]
```diff
@@ -21,6 +24,39 @@
 '''
 
 
+class DocGenThread(CalcThread):
+    """Thread performing the fit """
```
Comment (rozyczko) [inline src/sas/qtgui/Utilities/DocViewWidget.py:28]: Are those changes related to the issue of broken Linux testing? Or just a stealthy fix for doc gen? :)

File: installers/installer.iss [hunk 2]
```diff
@@ -26,7 +26,7 @@ UsedUserAreasWarning=no
 LicenseFile=license.txt
 ArchitecturesInstallIn64BitMode=x64
 OutputBaseFilename=setupSasView
-SetupIconFile=dist\sasview\_internal\images\ball.ico
+SetupIconFile=dist\sasview\images\ball.ico
```
Comment (rozyczko) [inline installers/installer.iss:29]: These fixes for `_internal` directory look like they belong to a separate issue/PR?

**Comment by krzywon** [comment]
@rozyczko, to answer both of your questions, this branch originated from waht is done in the pyinstaller fixes PR #2862. Without the changes from that branch, the Linux distro cannot launch. This is just more work on top of that.

The only new changes are in ci.yml.

**Comment by butlerpd** [comment]
Guessing we wait for #2862 to be approved/merged before merging this? on the other hand, the is `ball. ico` still the right file to be pulling in?

**Comment by lucas-wilkins** [comment]
> Guessing we wait for https://github.com/SasView/sasview/pull/2862 to be approved/merged before merging this? on the other hand, the is ball. ico still the right file to be pulling in?

Yes, it is

File: src/sas/qtgui/Utilities/DocViewWidget.py [hunk 3]
```diff
@@ -102,6 +138,7 @@ def regenerateIfNeeded(self):
         rst_path = MAIN_DOC_SRC
         base_path = self.source.parent.parts
         url_str = str(self.source)
+        create_user_files_if_needed()
 
         if not MAIN_DOC_SRC.exists() and not HELP_DIRECTORY_LOCATION.exists():
```
Comment (lucas-wilkins) [inline src/sas/qtgui/Utilities/DocViewWidget.py:143]: should this be an or?

**Review by lucas-wilkins — APPROVED** [review]
Approved assuming it works. Does this mean we can replace some of the worst stuff in the docs building system?

**Comment by wpotrzebowski** [comment]
I've tried on Rocky Linux. It starts without problem but it fails (seg fault) when trying to Load Data and at the end of generating documentation. 

Screenshot 2024-08-01 at 15 28 55

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -452,13 +452,14 @@ jobs:
...
-          sudo apt install libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0 x11-utils xvfb
+          sudo DEBIAN_FRONTEND=noninteractive apt install libegl1 libopengl0
+          sudo DEBIAN_FRONTEND=noninteractive apt install libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0 x11-utils xvfb
 
       - name: Try running the installation (Linux)
         if: ${{ startsWith(matrix.os, 'ubuntu' )}}
         # If sasview has crashed on its own, then this should return an error
         run: |
-          xvfb-run -a --server-args="-screen 0 1024x768x24" $RUN_SASVIEW &
+          xvfb-run -e /dev/stdout -a --server-args="-screen 0 1024x768x24" $RUN_SASVIEW &
```
Comment (llimeht) [inline .github/workflows/ci.yml:462]: My conclusion was that a lot more was needed here - I've been playing and have put that in PR #3070

File: installers/installer.iss [hunk 5]
```diff
@@ -55,9 +55,8 @@ end;
 
 [Files]
 Source: "dist\sasview\sasview.exe"; DestDir: "{app}"; Flags: ignoreversion
+Source: "dist\sasview\plugin_models\*"; DestDir: "{%USERPROFILE}\.sasview\plugin_models"
```
Comment (llimeht) [inline installers/installer.iss:58]: Is this a separate change for the windows installer?

Comment (krzywon) [reply installers/installer.iss:58]: This was branched off a previous branch that is already merged. The only changes unique to this branch are in `ci.yml`. For some reason, the remaining changes still appear.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (krzywon) [reply .github/workflows/ci.yml:462]: I think your work might supersede this PR, but I need to take a better look. If so, I'll close this.

**Comment by krzywon** [comment]
Superseded by #3070

**Pull request closed without merging**